extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/DevServicesElasticsearchProcessor.java

```diff
@@ -0,0 +1,180 @@
+    static final String DEV_SERVICE_LABEL = "quarkus-dev-service-elasticsearch";
+    static final int ELASTICSEARCH_PORT = 9200;
+
+    private static final ContainerLocator elasticsearchContainerLocator = new ContainerLocator(DEV_SERVICE_LABEL,
+            ELASTICSEARCH_PORT);
+
+    static volatile DevServicesResultBuildItem.RunningDevService devService;
+    static volatile ElasticsearchDevServicesBuildTimeConfig cfg;
+    static volatile boolean first = true;
+
+    private final IsDockerWorking isDockerWorking = new IsDockerWorking(true);
+
+    @BuildStep(onlyIfNot = IsNormal.class, onlyIf = GlobalDevServicesConfig.Enabled.class)
+    public DevServicesResultBuildItem startElasticsearchDevService(
+            LaunchModeBuildItem launchMode,
+            ElasticsearchDevServicesBuildTimeConfig configuration,
+            List<DevServicesSharedNetworkBuildItem> devServicesSharedNetworkBuildItem,
+            Optional<ConsoleInstalledBuildItem> consoleInstalledBuildItem,
+            CuratedApplicationShutdownBuildItem closeBuildItem,
+            LoggingSetupBuildItem loggingSetupBuildItem, GlobalDevServicesConfig devServicesConfig) {
+
+        if (devService != null) {
+            boolean shouldShutdownTheBroker = !configuration.equals(cfg);
+            if (!shouldShutdownTheBroker) {
+                return devService.toBuildItem();
+            }
+            shutdownElasticsearch();
+            cfg = null;
+        }
+
+        StartupLogCompressor compressor = new StartupLogCompressor(
+                (launchMode.isTest() ? "(test) " : "") + "Elasticsearch Dev Services Starting:",
+                consoleInstalledBuildItem, loggingSetupBuildItem);
+        try {
+            devService = startElasticsearch(configuration, launchMode,
+                    !devServicesSharedNetworkBuildItem.isEmpty(),
+                    devServicesConfig.timeout);
+            compressor.close();
+        } catch (Throwable t) {
+            compressor.closeAndDumpCaptured();
+            throw new RuntimeException(t);
+        }
+
+        if (devService == null) {
+            return null;
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
+            Runnable closeTask = () -> {
+                if (devService != null) {
+                    shutdownElasticsearch();
+                }
+                first = true;
+                devService = null;
+                cfg = null;
+            };
+            closeBuildItem.addCloseTask(closeTask, true);
+        }
+        cfg = configuration;
+
+        if (devService.isOwner()) {
+            log.infof(
+                    "Dev Services for Elasticsearch started. Other Quarkus applications in dev mode will find the "
+                            + "server automatically. For Quarkus applications in production mode, you can connect to"
+                            + " this by starting your application with -Dquarkus.elasticsearch.hosts=%s",
```

Comment:
Correct me if I'm wrong, but this message applies to the Elasticsearch clients only, doesn't it? This won't be enough to configure Hibernate Search, for example.

I don't see this kind of messages in other dev services, so I wonder if it's a good idea to include it. IMO it could help some users, but also confuse others who would wonder why this configuration doesn't work with Hibernate Search.

Reply:
Kafka and MongoDB have such log but I can remove it.
We definitly should use a build item so the extension using Elasticsearch would be able to customize this.